src/Raven.Server/Documents/PeriodicBackup/PeriodicBackupRunner.cs

```diff
@@ -173,29 +173,45 @@ public PeriodicBackupRunner(DocumentDatabase database, ServerStore serverStore,
             PeriodicBackupStatus backupStatus,
             string responsibleNodeTag,
             bool skipErrorLog = false)
+        {
+            return GetNextBackupDetailsLogic(configuration, backupStatus, responsibleNodeTag, skipErrorLog, _databaseWakeUpTimeUtc, _database, _serverStore.NodeTag, _logger);
+        }
+
+        private static NextBackup GetNextBackupDetailsLogic(
```

Comment:
Let's extract this logic to a separate class (`BackupUtils`)

Reply:
Also, move the relevant methods and use a delegate for error handling

Reply:
Done